chore: remove apple pay domain verification file

frosso opened the pull request:
Fixes #

#### Changes proposed in this Pull Request

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

From pbUcTB-6Hc-p2

It looks like we no longer need to add the verification file.
I am not deleting existing domain verification files, as proposed in the P2 above.

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

This is removing the requirement that creates a domain verification file at `/.well-known/apple-developer-merchantid-domain-association`.
You can verify this is still working fine by:
- Ensuring WooPayments is enabled on the site
- Ensuring the Google Pay/Apple Pay is disabled on the site (if you have it active, disable it)
- Navigate to the WooPayments settings page
- Enable Google Pay/Apple Pay
- Save the settings
- Navigate to the WooPayments logs (WooCommerce > Status > Logs)
- There should be a new entry "Your domain has been verified with Apple Pay!" (in case you use a "real" site)
  - Or, alternatively, an error from Stripe

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

frosso committed:
chore: remove apple pay domain verification file

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10733` or branch name `chore/remove-apple-pay-domain-verification-file` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=chore/remove-apple-pay-domain-verification-file) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 014d25d2b3495343cc55c3fbfd28c129b923b087
- Build time: 2025-05-21 09:42:46 UTC

_Note: the build is updated when a new commit is pushed to this PR._

frosso committed:
Merge branch 'develop' into chore/remove-apple-pay-domain-verification-file

frosso commented on includes/class-wc-payments-apple-pay-registration.php:
```diff
@@ -135,114 +129,11 @@ public function verify_domain_on_domain_name_change() {
 	 * Verify domain upon plugin update only in case the domain association file has changed.
 	 */
 	public function verify_domain_on_update() {
-		if ( $this->is_enabled() && ! $this->is_hosted_domain_association_file_up_to_date() ) {
```
I was uncertain about this. Before the changes, we were performing the verification only if the domain association file was not "up to date".
With these changes, we always perform the verification on upgrade (if WooPayments is enabled and Google Pay/Apple Pay are enabled).
I don't think it's a major issue?

htdat replied:
Based on 1e99d3aeb9674869c3b05c9a1e35ceb5a3e0bd41, I think it used to be a large issue when for every plugin update, we got bombarded from the client site to the `apple_pay/domains` and `payment_method_domains` endpoints. 

I read the PR introducing https://github.com/Automattic/woocommerce-payments/pull/1239 this `on_update` action, its issue #1071, and the respective Stripe issues/PRs, and I think that's it's safe to remove this method completely. The current `verify_domain_on_domain_name_change` is already enough to detect the domain change, run the `register_domain` again.

frosso replied:
Thank you for locating that and for the background info @htdat ! I removed the hook and method in bbc3aea7b 👍

frosso committed:
WIP

htdat reviewed (commented):
Following the manual tests look good.
I also tried to modify the domain name saved in `apple_pay_verified_domain` setting to trigger the `verify_domain_on_domain_name_change` action for the `init` hook, and it works well. 

However, see my comment above regarding `verify_domain_on_update`.

frosso committed:
remove verification call on plugin update

frosso committed:
Merge branch 'develop' into chore/remove-apple-pay-domain-verification-file

htdat reviewed (changes requested):
😬 sorry, I was too fast. It seems there is a PHP test failure. Could you check it?

frosso committed:
more tests

frosso commented:
Yes, thank you @htdat ! I made the changes and didn't run the tests locally! I now removed the test case (which was testing the no-longer-existing method) and added a few more test cases in 47381d2

frosso committed:
Merge branch 'develop' into chore/remove-apple-pay-domain-verification-file
